### DISCUSSION TITLE
some problem with showing picture on canvas

### DISCUSSION THREAD

**Discussion by Stink-Po** [TomSchimansky/CustomTkinter#1098] [General]
hello i try to make a simpel app for watermark photo when i try to make a image on canvas i get blank canvas with the code blow:

```
import customtkinter
from PIL import Image, ImageTk


my_font1 = ('times', 18, 'bold')
customtkinter.set_appearance_mode("dark")
customtkinter.set_default_color_theme("dark-blue")
window = customtkinter.CTk()
window.title("Water mark Photo")
window.config(padx=50, pady=50)
window.geometry("600x600")
input_pic_label = customtkinter.CTkLabel(master=window,
                                         text="Choose Your Photo",
                                         font=my_font1)
input_pic_label.grid(row=0, column=0, padx=(180, 0), pady=(200, 0))

choose_pic_button = customtkinter.CTkButton(master=window,
                                            text="Upload Photo",
                                            command=lambda: upload_file())
choose_pic_button.grid(row=1, column=0, padx=(180, 0))


def upload_file():
    choose_pic_button.destroy()
    input_pic_label.destroy()
    f_types = [('Jpg Files', '*.jpg'), ('Png Files', '*.png')]
    filename = customtkinter.filedialog.askopenfilename(filetypes=f_types)
    image = Image.open(filename)
    img = ImageTk.PhotoImage(Image.open(filename))
    ct = customtkinter.CTkImage(dark_image=image,
                                size=(500, 500))
    original_image_size = image.size
    print(original_image_size)
    resized_image = image.resize((500, 500))
    resized_image.save("temp.jpg")
    resized_image = ImageTk.PhotoImage(resized_image)
    canvas = customtkinter.CTkCanvas(master=window, width=500, height=500, highlightthickness=0, bg="black")
    canvas.create_image(250, 250, image=resized_image)
    canvas.grid(row=2, column=1, columnspan=2)
    window.update()
   # label = customtkinter.CTkLabel(master=window, text="", image=resized_image)


window.mainloop()
```

but when i un comment this line:

`label = customtkinter.CTkLabel(master=window, text="", image=resized_image)`

 event i dont grid it on screen i see the picture on canvas how ever i get this warning :

C:\Users\Pourya\PycharmProjects\watermark\venv\lib\site-packages\customtkinter\windows\widgets\core_widget_classes\ctk_base_class.py:182: UserWarning: CTkLabel Warning: Given image is not CTkImage but <class 'PIL.ImageTk.PhotoImage'>. Image can not be scaled on HighDPI displays, use CTkImage instead.

  warnings.warn(f"{type(self).__name__} Warning: Given image is not CTkImage but {type(image)}. " +

**Comment by TomSchimansky** [comment]
That's just a warning to tell you that you should prefer using CTkImages over ImageTk.PhotoImage, because the PhotoImage has no scaling, so on higher resolution displays the app layout will be ruined. Maybe PhotoImage will become deprecated at some point.

**Reply by Stink-Po** [reply]
I use CTKImage I don't see the warning now but I still don't see the image in canvas if I don't use the picture in a label or for etc a button
